code/application/client/src/UnitView.js

```diff
@@ -1,109 +1,103 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 
 import Card from 'react-bootstrap/Card';
 
-class UnitView extends React.Component {
-
-    constructor(props) {
-      super(props);
-      this.state = {
-        isLoaded: false,
-        isError: false,
-        item: {}
-      }
-    }
-  
-    _fetchData() {
-      const url = "http://localhost:8000/api/details/" + this.props.type + "/" + this.props.code;
-      console.log("UnitView fetch " + url);
-      fetch(url)
-      .then(r => {
-        if (!r.ok) {
-          const e = "Attempting to load " + r.url + " got status " + r.status + ".";
-          this.setState({isLoaded: true, isError: true, errmsg: e})
-        }
-        return r.json()
-      })
-      .then (
-        (result) => {
-          if (!this.state.isError) {
-            this.setState({isLoaded: true, item: result})
-          }
-        },
-        (error) => {
-          this.setState({isLoaded: true, isError: true, errmsg: "Network error"})
-        }
-      )
-    }
-
-    componentDidMount() {
-      this._fetchData();
-    }
+function UnitView({ displayName, type, code, callback, details, children}) {
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [item, setItem] = useState({});
+  const [errmsg, setErrmsg] = useState(null);
   
-    componentDidUpdate(oldProps) {
-      if (this.props.code !== oldProps.code) {
-        this.setState({isLoaded: false});
-        this._fetchData();
+  function _fetchData() {
+    const url = "http://localhost:8000/api/details/" + type + "/" + code;
+    console.log("UnitView fetch " + url);
+    fetch(url)
+    .then(r => {
+      if (!r.ok) {
+        const e = "Attempting to load " + r.url + " got status " + r.status + ".";
```

Comment:
You should use a format string:
```suggestion
const e = `Attempting to load ${r.url} got status ${r.status}.`;
```